### PR TITLE
Fix invalid error in wrong case

### PR DESCRIPTION
**Pull request by 30mar**
## Description
Fix invalid QRCode errors in case of empty or nil date of birth
Add new techspec changes 

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6632

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift [hunk 1]
```diff
@@ -41,7 +41,14 @@ enum Route {
 				self = .rapidAntigen( .failure(.invalidTestCode))
 				return
 			}
-
+			let allIsNil = testInformation.firstName == nil && testInformation.firstName == nil && testInformation.dateOfBirthString == nil
+			// dateOfBirth is nil if dateOfBirthString is nil OR if dateOfBirthString is invalid
+			let allIsValid = testInformation.firstName != nil && testInformation.firstName != nil && testInformation.dateOfBirth != nil
+			
+			guard allIsNil || allIsValid else {
+				self = .rapidAntigen( .failure(.invalidTestCode))
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:49]: Log error pls.

File: src/xcode/ENA/ENA/Source/Models/Submisstion/AntigenTestInformation.swift [hunk 2]
```diff
@@ -14,14 +14,19 @@ struct AntigenTestInformation: Codable, Equatable {
...
-	) {
+		) {
 		self.hash = hash
 		self.timestamp = timestamp
 		self.firstName = firstName
 		self.lastName = lastName
 		self.dateOfBirth = dateOfBirth
+		guard let dateOfBirth = dateOfBirth else {
+			dateOfBirthString = nil
+			return
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Models/Submisstion/AntigenTestInformation.swift:25]: Log something when return in guard

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Models/Submisstion/AntigenTestInformation.swift (hunk 2 above)
Comment (30mar) [reply src/xcode/ENA/ENA/Source/Models/Submisstion/AntigenTestInformation.swift:25]: but this is not an error, we just set the property accordingly

Comment (pascal-brause) [reply src/xcode/ENA/ENA/Source/Models/Submisstion/AntigenTestInformation.swift:25]: ah did not see because of github's reduction for review windows.

**Review by pascal-brause — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift [hunk 3]
```diff
@@ -39,9 +40,18 @@ enum Route {
 				  testInformation.timestamp >= 0
 			else {
 				self = .rapidAntigen( .failure(.invalidTestCode))
+				Log.error("Antigen test data is not, either timeStamp is -ve or the hash is invalid", log: .qrCode)
+				return
+			}
+			let allIsNil = testInformation.firstName == nil && testInformation.firstName == nil && testInformation.dateOfBirthString == nil
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:46]: I would rather check for `isEmpty`:
```suggestion
			let allIsNil = testInformation.firstName.isEmpty && testInformation.firstName.isEmpty && testInformation.dateOfBirthString.isEmpty
```

**Pull request merged**